packages/donate-button/src/generic-button-loader.tsx

```diff
@@ -0,0 +1,54 @@
+import {render} from 'preact';
+import {
+	GenericButtonProps,
+	DonateButtonOptions
+} from 'src/helpers/options-types';
+
+interface GenericButtonLoader {
+	selector: string;
+	onClick: () => void;
+	widgetOptions: Partial<DonateButtonOptions>;
+	options: Partial<GenericButtonProps>;
+}
+
+const getCharityName = (
+	widgetOptions: Partial<DonateButtonOptions>
+): string => {
+	const genericFoundation = 'your-foundation';
+	if (typeof widgetOptions.onSubmit === 'function') {
+		return genericFoundation;
+	}
+
+	return widgetOptions.onSubmit?.charity
+		? widgetOptions.onSubmit?.charity
+		: genericFoundation;
+};
+
+const genericButtonLoader = ({
+	selector,
+	options = {},
+	onClick,
+	widgetOptions
+}: GenericButtonLoader) => {
+	const div = document.querySelector(selector);
+
+	const charity = getCharityName(widgetOptions);
+	const hrefUrl = `https://www.every.org/${charity}/donate`;
```

Comment:
Also would be nice to re-use the url creation from what we actually use in the real donate button. If we do it like this for instance, it loses important information as a fallback for the event handler code not working, for instance we allow tying a "partner webhook token" that allows partners to receive updates when people attempt a donation; by not using the real URL, that query param goes missing, so this "link fallback" would not pass information back to the donor website.

I think we can just use the same URL construction function, and omit stuff that hasn't been selected like frequency and what not. A pro of this is that it makes the ability to use a staging environment simpler (since we'd only need to augment that function).